poly-commitment/src/ipa.rs

```diff
@@ -408,18 +408,36 @@ where
     ) -> PolyComm<G> {
         let is_zero = plnm.is_zero();
 
-        let coeffs: Vec<_> = plnm.iter().map(|c| c.into_bigint()).collect();
-
         // chunk while committing
-        let mut chunks = vec![];
-        if is_zero {
-            chunks.push(G::zero());
+        let mut chunks: Vec<_> = if is_zero {
+            vec![G::zero()]
```

Comment:
This breaks the invariant that the length of this vector is equal to `num_chunks`. Likewise below.